conf/jest/enzyme-adapter.js

```diff
@@ -11,3 +11,6 @@ Enzyme.configure({ adapter: new Adapter() });
 // make Enzyme functions available in all test files without importing
 global.shallow = shallow;
 global.mount = mount;
+
+// testing utility functions
+global.queryAllByTestId = (wrapper, testid) => wrapper.find(`[data-testid="${testid}"]`).hostNodes();
```

Comment:
Is `.hostNodes()` needed or can we return the wrappers themselves? I think we want to avoid interacting directly with DOM nodes when possible.

Reply:
I want to count only the actual rendered testids, not everywhere the prop is passed. The Enzyme wrappers would find the react node, any child component that passes props through, and final DOM node.

Reply:
Should we be using `render` rather than `mount`, in that case? If the final rendered markup is what we want?

Reply:
In this case a static render might work... but that doesn't work for anything stateful or that tests interactions, right? I thought this helper would be fairly useful in other types of tests.

Reply:
Possibly, but I think we should probably add global methods only when their value is very high. Having a helper to find/filter on `data-testid` attribute values seems valuable, but returning `hostNodes` less so. 

It also looks like the `.render` [helper](https://airbnb.io/enzyme/docs/api/ReactWrapper/render.html) may also be useful to accomplish this inline.

Reply:
https://github.com/box/box-ui-elements/pull/1027/commits/3c9e928ad94d86f97d86c7af5ced1897dd2f3431